infra/network-isolation.bicep

```diff
@@ -0,0 +1,202 @@
+              name: appServicePlanName
+              properties: {
+                serviceName: 'Microsoft.Web/serverFarms'
+              }
+            }
+          ]
+        }
+      }
+    ]
+  }
+}
+
+// DNSs Zones
+
+module blobDnsZone './core/networking/private-dns-zones.bicep' = {
+  name: 'blob-dnzones'
+  params: {
+    dnsZoneName: 'privatelink.blob.${environmentData.suffixes.storage}'
+    tags: tags
+    virtualNetworkName: vnet.outputs.name
+  }
+}
+
+module documentsDnsZone './core/networking/private-dns-zones.bicep' = {
+  name: 'documents-dnzones'
+  params: {
+    dnsZoneName: 'privatelink.documents.azure.com'
+    tags: tags
+    virtualNetworkName: vnet.outputs.name
+  }
+}
+
+module websitesDnsZone './core/networking/private-dns-zones.bicep' = {
+  name: 'websites-dnzones'
+  params: {
+    dnsZoneName: 'privatelink.azurewebsites.net'
+    tags: tags
+    virtualNetworkName: vnet.outputs.name
+  }
+}
+
+module cognitiveservicesDnsZone './core/networking/private-dns-zones.bicep' = {
+  name: 'cognitiveservices-dnzones'
+  params: {
+    dnsZoneName: 'privatelink.cognitiveservices.azure.com'
+    tags: tags
+    virtualNetworkName: vnet.outputs.name
+  }
+}
+
+module openaiDnsZone './core/networking/private-dns-zones.bicep' = {
+  name: 'openai-dnzones'
+  params: {
+    dnsZoneName: 'privatelink.openai.azure.com'
+    tags: tags
+    virtualNetworkName: vnet.outputs.name
+  }
+}
+
+module searchDnsZone './core/networking/private-dns-zones.bicep' = {
+  name: 'searchs-dnzones'
+  params: {
+    dnsZoneName: 'privatelink.search.windows.net'
+    tags: tags
+    virtualNetworkName: vnet.outputs.name
+  }
+}
+
+module storagePrivateEndpoint './core/networking/private-endpoint.bicep' = {
+  name: 'storageprivateendpoint'
+  params: {
+    location: location
+    name: '${abbrs.storageStorageAccounts}${abbrs.privateEndpoint}${resourceToken}'
+    tags: tags
+    subnetId: vnet.outputs.vnetSubnets[0].id
+    serviceId: storageAccountId
+    groupIds: [ 'blob' ]
+    dnsZoneId: blobDnsZone.outputs.id
+  }
+}
+
+module openAiPrivateEndpoint './core/networking/private-endpoint.bicep' = {
+  name: 'openaiprivateendpoint'
+  params: {
+    location: location
+    name: '${abbrs.cognitiveServicesAccounts}${abbrs.privateEndpoint}${resourceToken}'
+    tags: tags
+    subnetId: vnet.outputs.vnetSubnets[0].id
+    serviceId: openAiId
+    groupIds: [ 'account' ]
+    dnsZoneId: openaiDnsZone.outputs.id
+  }
+}
+
+module formRecognizerPrivateEndpoint './core/networking/private-endpoint.bicep' = {
+  name: 'formrecognizerprivateendpoint'
+  params: {
+    location: location
+    name: '${abbrs.cognitiveServicesFormRecognizer}${abbrs.privateEndpoint}${resourceToken}'
+    tags: tags
+    subnetId: vnet.outputs.vnetSubnets[0].id
+    serviceId: formRecognizerId
+    groupIds: [ 'account' ]
+    dnsZoneId: cognitiveservicesDnsZone.outputs.id
+  }
+}
+
+module searchStoragePrivateLink 'core/search/search-private-link.bicep' = {
```

Comment:
Are we sure this is needed? With our current architecture, the search service doesn't talk directly to blob store. I think this is only needed when using a blob indexer.